src/_pytest/junitxml.py

```diff
@@ -307,27 +323,22 @@ def record_xml_attribute(request):
     The fixture is callable with ``(name, value)``, with value being
     automatically xml-encoded
     """
-    from _pytest.warning_types import PytestExperimentalApiWarning, PytestWarning
+    from _pytest.warning_types import PytestExperimentalApiWarning
 
     request.node.warn(
         PytestExperimentalApiWarning("record_xml_attribute is an experimental feature")
     )
 
+    _warn_incompatibility_with_xunit2(request, "record_xml_attribute")
+
     # Declare noop
     def add_attr_noop(name, value):
         pass
 
     attr_func = add_attr_noop
-    xml = getattr(request.config, "_xml", None)
 
-    if xml is not None and xml.family != "xunit1":
-        request.node.warn(
-            PytestWarning(
-                "record_xml_attribute is incompatible with junit_family: "
-                "%s (use: legacy|xunit1)" % xml.family
-            )
-        )
-    elif xml is not None:
+    xml = getattr(request.config, "_xml", None)
+    if xml is not None:
```

Comment:
The false-branch is not covered.

Reply:
Nice, thanks for pointing that out. Added a new test. 👍